numpy/_core/_exceptions.py

```diff
@@ -88,10 +85,8 @@ def __str__(self):
         # only show the number if more than one input exists
         i_str = f"{self.in_i} " if self.ufunc.nin != 1 else ""
         return (
-            "Cannot cast ufunc {!r} input {}from {!r} to {!r} with casting "
```

Comment:
I thought the "missing" space in `input {}from` is an error, but it's actually required because the argument `i_str` itself ends with a space. Will fix this.